### ISSUE TITLE
Requests to download.finance.yahoo.com fail with "Sorry, Unable to process request at this time -- error 999".

### ISSUE THREAD

**Issue by darrylr02124** [yql/yql-tables#491]
Requests to the URL 
http://download.finance.yahoo.com/d/quotes.csv?f=aa2bb2b3b4cc1c3c4c6c8dd1d2ee1e7e8e9ghjkg1g3g4g5g6ii5j1j3j4j5j6k1k2k4k5ll1l2l3mm2m3m4m5m6m7m8nn4opp1p2p5p6qrr1r2r5r6r7ss1s7t1t7t8vv1v7ww1w4xy&s=AEO,BAC

result in the error "Sorry, Unable to process request at this time -- error 999."

This results in the failure of queries against the yahoo.finance.quotes table (e.g. https://query.yahooapis.com/v1/public/yql?q=select * from yahoo.finance.quotes where symbol in ("AEO", "GME")&diagnostics=true&env=store://datatables.org/alltableswithkeys).

**Comment by saunders77** [comment]
I'm hitting the same issue. I've heard from many others about this problem starting today. Looks like the quotes table is completely broken right now.

**Comment by darrylr02124** [comment]
Yes it does. And the frustrating thing is that the table admins never seem to acknowledge the issue; they simply fix it (eventually), and it "magically" begins to work again. The previous outage lasted for a week, but download.finance.yahoo.com was still working during that outage. The culprit was the outage of (at least) one of the downstream servers that converts the CSV to XML (which resulted in intermittent failures of queries against the yahoo.finance.quotes table).
 
In addition to posting here, which didn't seem to help (but might have), I tweeted to @ydn and @yql and emailed the yahooapis.com domain technical contact at "domainadmin at yahoo-inc.com". I'm not sure whether those actions helped, but they couldn't have hurt.

**Comment by Wingblur** [comment]
Maybe time to move to Google Finance? They are also real-time as opposed to being 15 mins delayed.

**Comment by aleufms** [comment]
Are there a way to use pm.finance table from yql?

http://developer.yahoo.com/yql/console/?q=select%20*%20from%20pm.finance%20where%20symbol%3D%22c%3Df%22

It works in site, but the standalone call: https://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20pm.finance%20where%20symbol%3D%22c%3Df%22&format=json&diagnostics=true&callback= don't

Their security is "APP" level instead, and I don't figure it out how to subscribe to get a key or auth session or whatever to access data. 

Inspecting the calls from website, it calls:  https://developer.yahoo.com/yql/console/proxy.php, and if i copy the crumb (from post body) and the cookie (from header) I can access the data.

**Comment by saunders77** [comment]
Looks like Yahoo may have intentionally disabled it: https://forums.yahoo.net/t5/Known-issues-and-updates/Finance-download-errors/m-p/387511#M13

**Comment by darrylr02124** [comment]
@Wingblur , where can I find doc on the Google Finance API? I googled it and found some outdated information about an interface that has now been deprecated, but haven't found anything current yet. Thnx!

**Comment by saunders77** [comment]
As far as I can tell Google hasn't had anything available since 2012 when they deprecated the Finance API that @darrylr02124 is referencing: https://developers.googleblog.com/2012/04/changes-to-deprecation-policies-and-api.html

But yeah, I'd love to be wrong there, @Wingblur

**Comment by darrylr02124** [comment]
@Wingblur, the following query still returns data:
https://finance.google.com/finance?q=AAPL&output=json
However, to @saunders77  point, I'm wondering whether it's still supported, or just working by happenstance.

**Comment by saunders77** [comment]
@darrylr02124, thanks, that's a good sign! I'd love to try some more queries at that endpoint. Is there still a link where that query is documented, or where you got it from?

**Comment by smittyxi** [comment]
I was a Y! quotes.csv user until it broke earlier this week and was unexpectedly shut down. Lots of frustration over on Y! uservoice here- https://yahoo.uservoice.com/forums/382977-finance/suggestions/32103877-http-download-finance-yahoo-com-d-quotes-csv-s-a

Perhaps it has to do with their agreements with the quotes service they use. Previously was using MSN quotes at http://moneycentral.msn.com/investor/external/excel/quotes.asp but that is gone too, as is the Google Finance API, but that Google JSON quote service looks promising. Just keep in mind, it might change soon. if you use certain features on google finanance, like editing a portfolio, it returns a message that says-

>  Google Finance is under renovation. As a part of this process, the Portfolios feature won't be available after mid-November 2017. To keep a copy, download your portfolio.

**Comment by darrylr02124** [comment]
@saunders77 , I found that query at https://stackoverflow.com/questions/46070126/google-finance-json-stock-quote-stopped-working . There are lots of caveats there that strongly suggest that this could go away at any time (due to the deprecation of Google Finance).

**Comment by 2121Savage** [comment]
I thought it was my Excel. But now Stock connector is getting "Yahoo's service is down. (see details)" this plugin was very helpful. Now I have to manually enter numbers. Any update?
